File: smite/src/main/java/hirez/smite/object/Item.java
package hirez.smite.object;

import com.fasterxml.jackson.annotation.JsonProperty;
import hirez.api.object.adapters.TextToBoolean;
import hirez.api.object.interfaces.ReturnedMessage;
import lombok.Data;

import java.util.List;

@Data
public class Item implements ReturnedMessage {

    @TextToBoolean
    @JsonProperty("ActiveFlag")
    private final boolean active;
    @JsonProperty("ChildItemId")
    private final long childId;
    @JsonProperty("DeviceName")
    private final String name;
    @JsonProperty("IconId")
    private final long iconId;
    @JsonProperty("ItemDescription")
    private final Description description;
    @JsonProperty("ItemId")
    private final long id;
    @JsonProperty("ItemTier")
    private final int itemTier;
    @JsonProperty("Price")
    private final int price;
    @JsonProperty("RestrictedRoles")
    private final String restrictedRoles;
    @JsonProperty("RootItemId")
    private final long rootItemId;
    @JsonProperty("ShortDesc")
    private final String shortDescription;
    @JsonProperty("StartingItem")
    private final boolean startingItem;
    @JsonProperty("Type")
    private final String type;
    @JsonProperty("itemIcon_URL")
    private final String iconURL;
    @JsonProperty("ret_msg")
    private final String returnedMessage;

    @Data
    public static class Description {
        @JsonProperty("Description")
        private final String description;
        @JsonProperty("Menuitems")
        private final List<Index> menuItems;
        @JsonProperty("SecondaryDescription")
        private final String secondaryDescription;

        @Data
        public static class Index {
            @JsonProperty("Description")
            private final String description;
            @JsonProperty("Value")
            private final String value;
        }
    }
}

File: api/src/main/kotlin/hirez/json/item.kt
package hirez.json

import com.google.gson.annotations.SerializedName

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Item(
			@SerializedName("ret_msg")
			override val returnMessage: String,
			val childItemId: Int,
			val deviceName: String,
			val iconId: Int,
			val itemDescription: ItemDescription,
			val itemId: Int,
			val itemTier: Int,
			val price: Int,
			val restrictedRoles: String,
			val rootItemId: Int,
			val shortDesc: String,
			val startingItem: Boolean,
			val type: String,
			val itemIconURL: String
) : ReturnMessage

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class ItemDescription(
			val description: String,
			val menuitems: List<Menuitem>,
			val secondaryDescription: String?
)

/**
 *
 * @author Damian Staszewski [damian@stachuofficial.tv]
 * @version %I%, %G%
 * @since 1.0
 */
data class Menuitem(
			val description: String,
			val value: String
)